angular-client/src/pages/charging-page/components/fault-display/fault-display.component.ts

```diff
@@ -19,40 +46,115 @@ export default class FaultDisplay {
 
   ngOnInit() {
     this.storage.get(IdentifierDataType.COMM_TIMEOUT_FAULT).subscribe((value) => {
-      this.addFault(value.values[0], 'Comm Timeout');
+      this.addFault(value.values[0], 'Comm Timeout', FaultType.Charger);
     });
 
     this.storage.get(IdentifierDataType.HARDWARE_FAILURE_FAULT).subscribe((value) => {
-      this.addFault(value.values[0], 'Hardware Failure');
+      this.addFault(value.values[0], 'Hardware Failure', FaultType.Charger);
     });
 
     this.storage.get(IdentifierDataType.OVER_TEMP_FAULT).subscribe((value) => {
-      this.addFault(value.values[0], 'Over Temp');
+      this.addFault(value.values[0], 'Over Temp', FaultType.Charger);
     });
 
     this.storage.get(IdentifierDataType.VOLTAGE_WRONG_FAULT).subscribe((value) => {
-      this.addFault(value.values[0], 'Voltage Wrong');
+      this.addFault(value.values[0], 'Voltage Wrong', FaultType.Charger);
     });
 
     this.storage.get(IdentifierDataType.WRONG_BAT_CONNECT_FAULT).subscribe((value) => {
-      this.addFault(value.values[0], 'Wrong Battery Connect');
+      this.addFault(value.values[0], 'Wrong Battery Connect', FaultType.Charger);
+    });
+
+    /**
+     * This is based on the shepard enum for faults:
+     * https://github.com/Northeastern-Electric-Racing/ShepherdBMS-2/blob/6eb3f863ed131a15bdf98665532cb7807bbd2920/Core/Inc/datastructs.h#L39
+     */
+    this.storage.get(IdentifierDataType.BMS_FAULTS).subscribe((value) => {
+      this.addFault(value.values[0], 'MASSIVE L, Should not be a fault hahaha', FaultType.BMS);
```

Comment:
what?

Reply:
Ya... I can make the addFault function a little less confusing.

Reply:
Basically BMS_FAULTS are only based on hex value, where other faults have a specific identifier (they are booleans)

Reply:
Not really worth explaining addFault() it's kinda weird anyways, will just make the function into two or something.